lib/services/local/presence.go

```diff
@@ -1700,23 +1704,45 @@ func FakePaginate(resources []types.ResourceWithLabels, req proto.ListResourcesR
 		PredicateExpression: req.PredicateExpression,
 	}
 
+	// afterMatchCount counts matches after we found the limit.
+	afterMatchCount := 0
+	// filterAll is a flag to continue matching even after we found limit,
+	// to determine the total count. This flag is only enabled on the first fetch.
+	filterAll := req.NeedTotalCount && req.StartKey == ""
```

Comment:
Why the check for `StartKey` if the caller has asked for the total count?

Reply:
you're right, the caller can toggle this flag. i changed to always return count if flag enabled